packages/web3-eth-contract/test/integration/contract_defaults.test.ts

```diff
@@ -18,61 +18,67 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 import { Web3BaseProvider } from 'web3-types';
 import { Contract } from '../../src';
 import { GreeterBytecode, GreeterAbi } from '../shared_fixtures/build/Greeter';
-import { getSystemTestProvider, getSystemTestAccounts } from '../fixtures/system_test_utils';
+import {
+	getSystemTestProvider,
+	createTempAccount,
+	itIf,
+	isIpc,
+} from '../fixtures/system_test_utils';
 
 describe('contract', () => {
 	describe('defaults', () => {
 		let contract: Contract<typeof GreeterAbi>;
 		let deployOptions: Record<string, unknown>;
 		let sendOptions: Record<string, unknown>;
-		let accounts: string[];
+		let acc: { address: string; privateKey: string };
 
 		beforeEach(async () => {
 			contract = new Contract(GreeterAbi, undefined, {
 				provider: getSystemTestProvider(),
 			});
-
-			accounts = await getSystemTestAccounts();
+			acc = await createTempAccount();
 
 			deployOptions = {
 				data: GreeterBytecode,
 				arguments: ['My Greeting'],
 			};
 
-			sendOptions = { from: accounts[0], gas: '1000000' };
+			sendOptions = { from: acc.address, gas: '1000000' };
 		});
 
 		describe('defaultAccount', () => {
-			it('should use "defaultAccount" on "Contract" level instead of "from"', async () => {
-				// eslint-disable-next-line prefer-destructuring
-				Contract.defaultAccount = accounts[0];
-
-				const receiptHandler = jest.fn();
-
-				// We didn't specify "from" in this call
-				await contract
-					.deploy(deployOptions)
-					.send({ gas: '1000000' })
-					.on('receipt', receiptHandler);
-
-				// We didn't specify "from" in this call
-				expect(receiptHandler).toHaveBeenCalledWith(
-					expect.objectContaining({ from: accounts[0] }),
-				);
-			});
+			itIf(!isIpc)(
```

Comment:
Sorry, why this test should not apply to `IPC`? And why it was working when it was applied to all including IPC?

And if there a good reason for that, could we please, have a small comment hereon  why this test should not apply to `IPC`?

Reply:
done

Reply:
fixed